src/components/Modal/LogoutModal.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { MouseEvent, useContext, useEffect } from "react";
+import { useCookies } from "react-cookie";
+import { Trans, useTranslation } from "react-i18next";
+import { withRouter } from "react-router-dom";
+import {
+  Button,
+  Col,
+  Media,
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  Row
+} from "reactstrap";
+import logoutLogo from "../../assets/img/logout.svg";
+import { AlertContext } from "../../context/alert-context";
+import { LogoutModalContext } from "../../context/logout-modal-context";
+import {
+  baseUrlBackendClient,
+  manageErrorReturnCodes
+} from "../../utils/api-utils";
+
+/*
+ * Modal shown at first login to prompt user to add personal mail
+ */
+export const LogoutModal = withRouter(props => {
+  /**
+   * react-i18next translation hook
+   */
+  const { t } = useTranslation();
+
+  const [cookies, , removeCookie] = useCookies(["sessionToken"]);
+  const alertContext = useContext(AlertContext);
+  const showGenericErrorAlert = () => {
+    alertContext.setAlert({
+      alertColor: "danger",
+      alertText: t("common.errors.genericError.500"),
+      showAlert: true
+    });
+  };
+  const logoutModalContext = useContext(LogoutModalContext);
+
+  const toggleLogoutModal = (isFromExpiredToken: boolean) => (
+    _: MouseEvent
+  ) => {
+    const isLogoutModalVisible = !logoutModalContext.logoutModal
+      .isLogoutModalVisible;
+    logoutModalContext.setLogoutModal({
+      isFromExpiredToken,
+      isLogoutModalVisible
+    });
+  };
+
+  useEffect(() => {
+    if (logoutModalContext.logoutModal.isFromExpiredToken) {
+      setTimeout(() => {
+        logoutModalContext.setLogoutModal({
+          isFromExpiredToken: false,
+          isLogoutModalVisible: false
+        });
+        removeCookie("sessionToken", { path: "/" });
+        props.history.push("/home");
+      }, 8000);
+    }
+  }, [logoutModalContext.logoutModal.isFromExpiredToken]);
+
+  const logout = () => {
+    baseUrlBackendClient(cookies.sessionToken)
+      .logout({})
+      .then(response => {
+        logoutModalContext.setLogoutModal({
+          isFromExpiredToken: false,
+          isLogoutModalVisible: false
+        });
+        if (response.isRight()) {
+          const respValue = response.value;
+          if (respValue.status === 204) {
+            removeCookie("sessionToken", { path: "/" });
+            props.history.push("/home");
+          } else {
+            const alertText = t(`common.errors.logout.${respValue.status}`)
```

Comment:
you can short circuit this to avoid building the same string twice:

```ts
 const alertText = t(`common.errors.logout.${respValue.status}`) || 
    t(`common.errors.logout.${respValue.status}`);

```